subsys/bluetooth/mesh/shell/shell_prop_cli.c

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdlib.h>
+#include <bluetooth/mesh/models.h>
+#include <shell/shell.h>
+
+#include "mesh/net.h"
+#include "mesh/access.h"
+#include "shell_utils.h"
+#include "common/log.h"
+
+static struct bt_mesh_model *mod;
+
+static void props_print(const struct shell *shell, int err, struct bt_mesh_prop_list rsp)
+{
+	if (!err) {
+		shell_print(shell, "ID count: %d", rsp.count);
+		for (uint8_t i = 0; i < rsp.count; i++) {
+			shell_print(shell, "ID %d val: %d", i, rsp.ids[i]);
+		}
+	}
+}
+
+static int shell_prop_client_props_get(const struct shell *shell, size_t argc, char *argv[])
+{
+	uint16_t id = (uint16_t)strtol(argv[1], NULL, 0);
+
+	if (!mod && !shell_model_first_get(BT_MESH_MODEL_ID_GEN_PROP_CLI, &mod)) {
+		return -ENODEV;
+	}
+
+	struct bt_mesh_prop_cli *cli = mod->user_data;
+	struct bt_mesh_prop_list rsp;
+
+	int err = bt_mesh_prop_cli_client_props_get(cli, NULL, id, &rsp);
+
+	props_print(shell, err, rsp);
+	return err;
+}
+
+static int shell_prop_props_get(const struct shell *shell, size_t argc, char *argv[])
+{
+	enum bt_mesh_prop_srv_kind kind = (enum bt_mesh_prop_srv_kind)strtol(argv[1], NULL, 0);
+
+	if (!mod && !shell_model_first_get(BT_MESH_MODEL_ID_GEN_PROP_CLI, &mod)) {
+		return -ENODEV;
+	}
+
+	struct bt_mesh_prop_cli *cli = mod->user_data;
+	struct bt_mesh_prop_list rsp;
+
+	int err = bt_mesh_prop_cli_props_get(cli, NULL, kind, &rsp);
+
+	props_print(shell, err, rsp);
+	return err;
+}
+
+static void prop_val_print(const struct shell *shell, int err, struct bt_mesh_prop_val rsp)
+{
+	if (!err) {
+		shell_print(shell, "Property value: %s", bt_hex(rsp.value, rsp.size));
+	}
+}
+
+static int shell_prop_prop_get(const struct shell *shell, size_t argc, char *argv[])
+{
+	enum bt_mesh_prop_srv_kind kind = (enum bt_mesh_prop_srv_kind)strtol(argv[1], NULL, 0);
+	uint16_t id = (uint16_t)strtol(argv[2], NULL, 0);
+
+	if (!mod && !shell_model_first_get(BT_MESH_MODEL_ID_GEN_PROP_CLI, &mod)) {
+		return -ENODEV;
+	}
+
+	struct bt_mesh_prop_cli *cli = mod->user_data;
+	struct bt_mesh_prop_val rsp;
```

Comment:
Here and below: `rsp` has to be initialized first to be filled with response:
https://github.com/nrfconnect/sdk-nrf/blob/6cecce46c17b682fdac1b829caab76236b1ea2dd/subsys/bluetooth/mesh/gen_prop_cli.c#L106-L112

Reply:
Like this?: `struct bt_mesh_prop_val rsp = { .size = CONFIG_BT_MESH_PROP_MAXSIZE };`

Reply:
`value` should fit `size` number of ids.